src/modules/fancyzones/editor/FancyZonesEditor/GridEditor.xaml.cs

```diff
@@ -137,80 +137,94 @@ private void ExtendRangeToHaveEvenCellEdges()
             // extend each edge of the [(_startCol, _startRow) - (_endCol, _endRow)] range based on merged cells until you have 4 straight edges with no "straddling cells"
             GridLayoutModel model = Model;
 
-            while (_startRow > 0)
+            // As long as there is an edge of the 2D range such that some zone crosses its boundary, extend that boundary.
+            // A single pass is not enough, a while loop is needed. This results in the unique smallest rectangle containing
+            // the initial range such that no zone is "broken".
```

Comment:
As code would be self-explaining now. Comments would be needless

Reply:
The whole code has been simplified, do you mind taking another look?